src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/indexpolicy/TransportIndexPolicyAction.kt

```diff
@@ -79,6 +96,41 @@ class TransportIndexPolicyAction @Inject constructor(
             }
         }
 
+        private fun checkTemplate(indexPatterns: List<String>, priority: Int) {
+            val possibleEx = validateFormat(indexPatterns)
+            if (possibleEx != null) {
+                actionListener.onFailure(possibleEx)
+                return
+            }
+
+            val searchRequest = SearchRequest()
+                .source(
+                    SearchSourceBuilder().query(
+                    QueryBuilders.existsQuery(ISM_TEMPLATE_FIELD)))
+                .indices(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX)
+
+            client.search(searchRequest, object : ActionListener<SearchResponse> {
+                override fun onResponse(response: SearchResponse) {
+                    val ismTemplates = ismTemplatesFromSearchResponse(response, xContentRegistry)
+                    val overlaps = findConflictingISMTemplates(request.policyID, indexPatterns, priority, ismTemplates)
```

Comment:
Just curious why is this is an issue - can we not assume if the priorities match for two policies containing matching indices the latest one is preferred since customer created it later?

Reply:
I think this is just by design, we don't allow this. 
From the very beginning, we try to duplicate the usage of index template. Index template won't allow this.